src/ripple/rpc/impl/KeypairForSignature.cpp

```diff
@@ -65,31 +65,43 @@ std::pair<PublicKey, SecretKey>
 keypairForSignature (Json::Value const& params, Json::Value& error)
 {
     bool const has_key_type  = params.isMember (jss::key_type);
-    bool const hasPassphrase = params.isMember (jss::passphrase);
-    bool const hasSecret     = params.isMember (jss::secret);
-    bool const hasSeed       = params.isMember (jss::seed);
-    bool const hasSeedHex    = params.isMember (jss::seed_hex);
 
-    int const n_secrets =
-        (hasPassphrase ? 1 : 0) +
-        (hasSecret ? 1 : 0) +
-        (hasSeed ? 1 : 0) +
-        (hasSeedHex ? 1 : 0);
+    // All of the signature types we allow.  Only one, though!
+    static constexpr char const* const sigTypes[]
+    {
+        jss::passphrase.c_str(),
+        jss::secret.c_str(),
+        jss::seed.c_str(),
+        jss::seed_hex.c_str()
+    };
+
+    // Identify which signature type is in use.
+    constexpr char noSig[] = "no signature";
+    char const* sigType = noSig;
+    int sigTypeCount = 0;
+    for (auto t : sigTypes)
+    {
+        if (params.isMember (t))
+        {
+            ++sigTypeCount;
+            sigType = t;
+        }
+    }
 
-    if (n_secrets == 0)
+    if (sigTypeCount == 0 || sigType == noSig)
```

Comment:
I originally wrote the code using `nullptr`.  I replaced the `nullptr` with `noSig` as a belt-and-suspenders technique with an eye toward maintenance bugs.  I use `sigType` later in the code as a `char*`.  If future maintenance messed up then we don't end up dereferencing a `nullptr`, which is dangerous.  We have a legit pointer to a descriptive string.  Personally, I slightly prefer the `noSig` approach.  But I'm open to disagreements.